src/common/context/observation-context-manager.js

```diff
@@ -0,0 +1,55 @@
+import { gosNREUM } from '../window/nreum'
+import { bundleId } from '../ids/bundle-id'
+import { EventContext } from './event-context'
+
+export class ObservationContextManager {
+  // These IDs are provided for backwards compatibility until the agent is fully updated
+  // use the observation context class.
+
+  static contextId = `nr@context:${bundleId}`
+  static contextOriginalId = `nr@original:${bundleId}`
+  static contextWrappedId = `nr@wrapped:${ObservationContextManager.contextId}`
+
+  static getObservationContextByAgentIdentifier (agentIdentifier) {
+    const nr = gosNREUM()
+    return Object.keys(nr?.initializedAgents || {}).indexOf(agentIdentifier) > -1
+      ? nr.initializedAgents[agentIdentifier].observationContext
+      : undefined
+  }
+
+  /**
+   * @type {WeakMap<WeakKey, {[key: string]: unknown}>}
+   */
+  #observationContext = new WeakMap()
+
+  /**
+   * Returns the observation context tied to the supplied construct. If there has been
+   * no observation construct created, an empty object is created and stored as the current
+   * context.
+   * @param key {unknown} The construct being observed such as an XHR instance
+   * @return {EventContext} An object of key:value pairs to track as
+   * part of the observation
+   */
+  getCreateContext (key) {
```

Comment:
The event emitter is still responsible for creating the observation context. I am not sure that is the right place for it and it's more just an easy central place to do it.